WSRequestHandler.cpp

```diff
@@ -1343,6 +1345,287 @@ void WSRequestHandler::HandleSetSceneItemCrop(WSRequestHandler* req) {
+            if (Utils::IsValidAlignment(alignment)) {
+                obs_sceneitem_set_alignment(scene_item, alignment);
+            }
+            // Send an error in the else statement?
+            // Append an error message to the response?
+        }
+        obs_sceneitem_set_pos(scene_item, &new_position);
+    }
+
+    if (req->hasField("rotation")) {
+        obs_sceneitem_set_rot(scene_item, (float)obs_data_get_double(req->data, "rotation"));
+    }
+
+    if (req->hasField("scale")) {
+        vec2 old_scale;
+        obs_sceneitem_get_scale(scene_item, &old_scale);
+        obs_data_t* req_scale = obs_data_get_obj(req->data, "crop");
+        vec2 new_scale = old_scale;
+        if (obs_data_has_user_value(req_scale, "x")) {
+            new_scale.x = obs_data_get_double(req_scale, "x");
+        }
+        if (obs_data_has_user_value(req_scale, "y")) {
+            new_scale.y = obs_data_get_double(req_scale, "y");
+        }
+    }
+
+    if (req->hasField("crop")) {
+        obs_sceneitem_crop old_crop;
+        obs_sceneitem_get_crop(scene_item, &old_crop);
+        obs_data_t* req_crop = obs_data_get_obj(req->data, "crop");
+        obs_sceneitem_crop new_crop = old_crop;
+        if (obs_data_has_user_value(req_crop, "top")) {
+            new_crop.top = obs_data_get_int(req_crop, "top");
+        }
+        if (obs_data_has_user_value(req_crop, "right")) {
+            new_crop.right = obs_data_get_int(req_crop, "right");
+        }
+        if (obs_data_has_user_value(req_crop, "bottom")) {
+            new_crop.bottom = obs_data_get_int(req_crop, "bottom");
+        }
+        if (obs_data_has_user_value(req_crop, "left")) {
+            new_crop.left = obs_data_get_int(req_crop, "left");
+        }
+        obs_sceneitem_set_crop(scene_item, &new_crop);
+    }
+
+    if (req->hasField("visible")) {
+        obs_sceneitem_set_visible(scene_item, obs_data_get_bool(req->data, "visible"));
+    }
+
+    if (req->hasField("bounds")) {
+        // TODO: figure out what to do with bounds type
+        vec2 old_bounds;
+        obs_sceneitem_get_bounds(scene_item, &old_bounds);
+        obs_data_t* req_bounds = obs_data_get_obj(req->data, "bounds");
+        vec2 new_bounds = old_bounds;
+        if (obs_data_has_user_value(req_bounds, "x")) {
+            new_bounds.x = obs_data_get_double(req_bounds, "x");
+        }
+        if (obs_data_has_user_value(req_bounds, "y")) {
+            new_bounds.y = obs_data_get_double(req_bounds, "y");
+        }
+        obs_sceneitem_set_bounds(scene_item, &new_bounds);
+        if (obs_data_has_user_value(req_bounds, "alignment")) {
+            const uint32_t bounds_alignment = obs_data_get_int(req_bounds, "alignment");
+            if (Utils::IsValidAlignment(bounds_alignment)) {
+                obs_sceneitem_set_bounds_alignment(scene_item, bounds_alignment);
+            }
+            // Send an error in the else statement?
+            // Append an error message to the response?
```

Comment:
Thoughts on what to do when alignment has flags for `left` and `right` ? this is invalid so we should probably send an error message

Reply:
It should send an error, because it's confusing for users to fail silently if the request is malformed.